src/lib/components/ui/TextField/TextField.jsx

```diff
@@ -6,7 +6,31 @@ import styles from './TextField.scss';
 
 export const TextField = (props) => {
   const SMALL_INPUT_SIZE = 10;
-  const HAS_SMALL_INPUT = props.inputSize && props.inputSize <= SMALL_INPUT_SIZE;
+
+  const setCustomInputSizeByType = (inputType, inputSize, max) => {
+    if (inputType === 'number' && max) {
+      return max.toString().length;
+    }
+
+    if (inputSize) {
```

Comment:
I still think about this. I would expect to have this condition as the first one. Reason is that you want to specify `max` attribute that can be entered into the field but at the same time, you could want to display only part of those numbers.

E.g. Phone number for Czech Republic can have 9 digits, so you set inputSize=9, but you allow to have max=9999999999999, because it is maximum value of phone number specified by IETF standards (i hope its IETF)

Reply:
I was also thinking about it for a while but eventually I could not figure out any use case. For the phone number you are mentioning I would use the `tel` input type along with the `pattern` attribute. Similarly, the card number is also rather a text string that happens to consist of numbers but it actually isn't a number (`Nan`!). I'm covering this in the docs (#15).